pkg/controller/template/render.go

```diff
@@ -260,23 +261,32 @@ func transpileToIgn(files, units []string) (*igntypes.Config, error) {
 
 	// Convert data to Ignition resources
 	for _, d := range files {
-		f := new(igntypes.File)
-		if err := yaml.Unmarshal([]byte(d), f); err != nil {
+		var f igntypes.File
+		if err := yaml.Unmarshal([]byte(d), &f); err != nil {
 			return nil, fmt.Errorf("failed to unmarshal file into struct: %v", err)
 		}
+		regdu := dataurl.New([]byte(*f.Contents.Source), "text/plain")
+		regdu.Encoding = dataurl.EncodingASCII
+		regduString := regdu.String()
+		f.Contents.Source = &regduString
```

Comment:
this needs also `f.Overwrite` to be set to `true` since spec v2 was `true` by default, but v3 is now false so to keep compatibility, we set it to always `true` for files (units are fine)

Reply:
done